roles/volume_create_delete/tasks/main.yml

```diff
@@ -1,16 +1,48 @@
 ---
 # tasks file for roles/volume_create_delete
 
+- name: Set Fact if volume_type is not defined
+  set_fact:
+    volume_type: false
+  when: volume_type is not defined
+
 - block:
-  - name: "Create Volume in {{ availability_zone }}"
+  - name: "Create SSD Volume in {{ availability_zone }}"
+    when: volume_type == "SSD"
+    openstack.cloud.volume:
+      state: present
+      availability_zone: "{{ availability_zone | default(omit) }}"
+      size: 10
+      volume_type: "{{ volume_type }}"
+      display_name: "{{ volume_name }}"
+    tags:
+      - "metric=create_volume"
+      - "metric=create_volume_SSD"
+    register: volume
+
+  - name: "Create SAS Volume in {{ availability_zone }}"
+    when: volume_type == "SAS"
     openstack.cloud.volume:
       state: present
       availability_zone: "{{ availability_zone | default(omit) }}"
       size: 10
+      volume_type: "{{ volume_type }}"
       display_name: "{{ volume_name }}"
     tags:
       - "metric=create_volume"
+      - "metric=create_volume_SAS"
     register: volume
+
+
+  - name: "Create Volume in {{ availability_zone }}"
+    when: volume_type == false
```

Comment:
remove  == false
as it's useless complexity
keep only when: volume_type

https://ansible.readthedocs.io/projects/lint/rules/literal-compare/

Reply:
corrected